styleguide.md

```diff
@@ -0,0 +1,180 @@
+For example, compare:
+- If equations are possible to simplify if their condition can be evaluated during translation.
+- `if`-equations are possible to simplify if their condition can be evaluated during translation.
+
+### Expressions
+
+Different constructs with _expression_ and _call_:
+
+Appearance | LaTeX source | Comment
+--- | --- | ---
+`if`-expression | `\lstinline!if!-expression` | Generic language concept
+`DynamicSelect`-expression | `\lstinline!DynamicSelect!-expression` | Currently with space in the document!
+`Curve`-expression | `\lstinline!Curve!-expression` | Currently with space in the document!
+`Real` expression | `\lstinline!Real! expression` | An expression of type `Real`
+`y` expression | `\lstinline!y! expression` | Expression for something named `y`
+`convertElement` call | `\lstinline!convertElement! call` | A call expression with callee `convertElement`
+
+Note: There is no need for hyphenation of "`convertElement` call" since we don't say "`Real` call" for a call expression of type `Real` (we have "`Real` expression" for this purpose).
+
+### The keywords themselves
+
+When referencing a keyword itself, hyphenation is not used, and when possible, a better describing word than _keyword_ is used:
+
+Appearance | LaTeX source | Comment
+--- | --- | ---
+the `input` prefix | `\lstinline!prefix! prefix` | For more prefixes, see `class-prefixes`, `base-prefix`, and `type-prefix` in the grammar.
+the `final` modifier | `the \lstinline!final! modifier` |
+the `each` keyword | `the \lstinline!each! keyword` |
+
+Depending on context, one can also swap the order, or drop the describing word completely:
+
+Appearance | LaTeX source
+--- | ---
+the keyword `each` | `the keyword \lstinline!each!`
+declared as `final` | `declared as \lstinline!final!`
+
+### Named functions and operators
+
+When referencing a named function of operator, hyphenation is not used, and it is common to not combine with any describing word:
+
+Appearance | LaTeX source
+--- | ---
+the `smooth` operator | `the \lstinline!smooth! operator`
+where `pre` is not explicitly used | `where \lstinline!pre! is not explicitly used`
+
+### Other things, special cases
+
+Incomplete list of various terminology with special formatting rules:
+
+Appearance | LaTeX source | Comment
+--- | --- | ---
+start value | `start value` | Value of the `start`-attribute (there could be exceptions!)
+reduction expression | `reduction expression` |
+base class | `base class` | Similarly: derived class
+base-clock | `base-clock` | Similarly: sub-cock
+
+## Definitions
+
+New terminology is either introduced with a `definition` environment, or as part of the running text.
+When part of the running text, the introduced terminology should be marked with `\firstuse` at the point of the definition.
+(The use of `\firstuse` helps us both produce consistent formatting and keep track of things that should appear in the document index.)
```

Comment:
Should `\firstuse` be followed by `\index`?

Reply:
I would say that should normally be the case.
The alternative would be to modify firstuse to do that automatically and add that information to the style-guide (to avoid double-index).

Reply:
My experience has been that the form used with `\firstuse` isn't always the form one wants in the index.  I added text saying that, as a general rule, `\firstuse` should be used in combination with `\index`.  OK?